general/g.mapsets/main.c

```diff
@@ -331,9 +330,12 @@ void append_mapset(char **path, const char *mapset)
     int len = (*path == NULL ? 0 : strlen(*path));
 
     *path = (char *)G_realloc(*path, len + strlen(mapset) + 2);
-    if (!len)
+    if (!len) {
         *path[0] = '\0';
-    strcat(*path, mapset);
-    strcat(*path, " ");
+        strcat(*path, mapset);
+    } else {
+        strcat(*path, " ");
```

Comment:
Path here be string with following correct format e.g. `"MAPSET MAPSET MAPSET"` (which is correct input string for tokenize string),
https://github.com/OSGeo/grass/blob/6253da1bd6ce48d23419e99e8b503edf46178490/general/g.mapsets/main.c#L274

not `"MAPSET MAPSET  MAPSET "` with additional  empty space on the end of the string, which causes a new empty line to be written to the SEARCH_PATCH file.